cisco_aci/datadog_checks/cisco_aci/tags.py

```diff
@@ -0,0 +1,151 @@
+                if ip:
+                    endpoint_meta.append("ip:" + ip)
+                mac = meta_attrs.get('mac')
+                if mac:
+                    endpoint_meta.append("mac:" + mac)
+                encap = meta_attrs.get('encap')
+                if encap:
+                    endpoint_meta.append("encap:" + encap)
+                # adding application tags
+        endpoint_meta += application_meta
+
+        context_hash = hash_mutable(endpoint_meta)
+        eth_meta = []
+        if self.tenant_tags.get(context_hash):
+            eth_meta = self.tenant_tags.get(context_hash)
+        else:
+            # adding eth and node tags
+            eth_list = self.api.get_eth_list_for_epg(tenant_name, app_name, epg_name)
+            for eth in eth_list:
+                eth_attrs = eth.get('fvRsCEpToPathEp', {}).get('attributes', {})
+                port = re.search('/pathep-\[(.+?)\]', eth_attrs.get('tDn', ''))
+                if not port:
+                    continue
+                eth_tag = 'port:' + port.group(1)
+                if eth_tag not in eth_meta:
+                    eth_meta.append(eth_tag)
+                node = re.search('/paths-(.+?)/', eth_attrs.get('tDn', ''))
+                if not node:
+                    continue
+                eth_node = 'node_id:' + node.group(1)
+                if eth_node not in eth_meta:
+                    eth_meta.append(eth_node)
+                # populating the map for eth-app mapping
+
+                tenant_fabric_key = node.group(1) + ":" + port.group(1)
+                if tenant_fabric_key not in self.tenant_farbic_mapper:
+                    self.tenant_farbic_mapper[tenant_fabric_key] = application_meta
+                else:
+                    self.tenant_farbic_mapper[tenant_fabric_key].extend(application_meta)
+
+                self.tenant_farbic_mapper[tenant_fabric_key] = list(set(self.tenant_farbic_mapper[tenant_fabric_key]))
+
+        tags = tags + endpoint_meta + eth_meta
+        if len(eth_meta) > 0:
+            self.log.debug('adding eth level tags: %s' % eth_meta)
+        return tags
+
+    def get_tags(self, obj, obj_type):
+        tags = []
+        if obj_type == 'endpoint_group':
+            tags = self.tenant_mapper(obj)
+        if obj_type == 'tenant':
+            tags = ["tenant:" + obj]
+        if obj_type == 'application':
+            tags = self.app_tags(obj)
+        return tags
+
+    def get_fabric_tags(self, obj, obj_type):
+        tags = []
+        obj = helpers.get_attributes(obj)
+        if obj_type == 'fabricNode':
+            if obj.get('role') != "controller":
+                tags.append("switch_role:" + obj.get('role'))
+            tags.append("apic_role:" + obj.get('role'))
+            tags.append("node_id:" + obj.get('id'))
+            tags.append("fabric_state:" + obj.get('fabricSt'))
+            tags.append("fabric_pod_id:" + helpers.get_pod_from_dn(obj.get('dn')))
```

Comment:
Can we merge these appends?

Reply:
No, it makes the line too long

Reply:
I got rid of the history but it used to be like that